tools/figma-inspector/src/utils/utils.ts

```
@@ -0,0 +1,35 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: MIT
+import { dispatchTS } from "./bolt-utils.js";
+
+export function writeTextToClipboard(str: string) {
+    const prevActive = document.activeElement;
+    const textArea = document.createElement("textarea");
+
+    textArea.value = str;
+
+    textArea.style.position = "fixed";
+    textArea.style.left = "-999999px";
+    textArea.style.top = "-999999px";
+
+    document.body.appendChild(textArea);
+
+    textArea.focus();
+    textArea.select();
+
+    return new Promise<void>((res, rej) => {
+        document.execCommand("copy") ? res() : rej();
+        textArea.remove();
+
+        if (prevActive && prevActive instanceof HTMLElement) {
+            prevActive.focus();
+        }
+    });
+}
+
+export function copyToClipboard(slintProperties: string) {
+    writeTextToClipboard(slintProperties);
+    dispatchTS("copyToClipboard", {
+        result: true,
+    });
```

Since writeTextToClipboard returns a promise, shouldn't we await here and mark copyToClipbard async? Or is it fine that dispatchTS() is called before the execCommand completed?

---

Figma is kind of like a browser and every plugin is sort of like an approved extension. Maybe because of that a bad extension could read the clipboard and steal sensitive data? So Figma have blocked use of normal clipboard API's. However it's still possible to have a copy to clipboard option as this deprecated web api can still be used. So I just cross my fingers and presume it worked. I updated it to now have some error handling. It's possible that Figma remove this or even Chrome removes it? At least now it won't fail silently when that happens.